lib/workload.hpp

```diff
@@ -232,49 +282,27 @@ class Comp {
     bool is_all = false;
 
     void normalize();
-
-    friend ostream& operator<<(ostream& os, const Comp& spec);
-    friend ostream& operator<<(ostream& os, const Comp* spec);
-    friend bool operator==(const Comp& spec1, const Comp& spec2);
-    friend bool operator!=(const Comp& spec1, const Comp& spec2);
-    friend bool operator<(const Comp& spec1, const Comp& spec2);
 };
 
-//************************************* WlSpec *************************************//
-typedef variant<Comp, Same, Incr, Decr, Unique> wl_spec_t;
-
-bool wl_spec_applies_to_queue(wl_spec_t spec, unsigned int queue);
-
-bool wl_spec_is_empty(wl_spec_t spec);
-
-bool wl_spec_is_all(wl_spec_t spec);
-
-unsigned int wl_spec_ast_size(const wl_spec_t wl_spec);
-
-ostream& operator<<(ostream& os, const wl_spec_t& wl_spec);
-
-bool operator==(const wl_spec_t& wl_spec1, const wl_spec_t& wl_spec2);
-bool operator<(const wl_spec_t& wl_spec1, const wl_spec_t& wl_spec2);
-
 //************************************* Timed WlSpec *************************************//
 
 class TimedSpec {
 
 public:
-    TimedSpec(wl_spec_t wl_spec, time_range_t time_range, unsigned int total_time);
-    TimedSpec(wl_spec_t wl_spec, unsigned int until_time, unsigned int total_time);
+    TimedSpec(std::shared_ptr<WlSpec> wl_spec, time_range_t time_range, unsigned int total_time);
```

Comment:
Why using shared_ptr?

Reply:
Will change all shared pointers to raw pointers 👍

Reply:
[Replace shared pointers w/ raw pointers](https://github.com/all-things-networking/fperf/pull/27/commits/4d70489742aa4f71d4d7c4c465ed19c64469c6ba)